### PR TITLE
[RFR] Make overring the title easier

### PR DESCRIPTION
**Pull request by djhi**
- [x] Easier title override using HOC `withAppTitle`
- [x] Easier manual testing by installing webpack and other sample dependencies in dev deps

### CODE REVIEW

**Comment by Kmaschta** [comment]
:+1:

File: src/Admin.js [hunk 1]
```diff
@@ -10,11 +10,12 @@ import crudSaga from './sideEffect/saga';
 import CrudRoute from './CrudRoute';
 import Layout from './mui/layout/Layout';
 
-const Admin = ({ restClient, appLayout = Layout, dashboard, children }) => {
+const Admin = ({ restClient, appLayout = Layout, dashboard, children, title = 'Admin on REST' }) => {
     const resources = React.Children.map(children, ({ props }) => props);
     const firstResource = resources[0].name;
     const sagaMiddleware = createSagaMiddleware();
     const reducer = combineReducers({
+        ui: () => ({ title }), // NOTE: maybe this could be a good fit for redux-ui ?
```
Comment (fzaninotto) [inline src/Admin.js]: I don't think we should use state for that. It's not going to change.

I understand that it's hard to pass to the Layout component, but I prefer that we use a HOC just for that.

Comment (djhi) [inline src/Admin.js]: Agreed, composition FTW !

File: src/mui/layout/Layout.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ import Menu from './Menu';
 injectTapEventPlugin();
 
 const Layout = ({ isLoading, children, route, title }) => {
+  console.log({ isLoading, route, title })
```
Comment (fzaninotto) [inline src/mui/layout/Layout.js]: `console.log()` spotted

File: example/app.js [hunk 3]
```diff
@@ -18,10 +20,12 @@ restServer.toggleLogging(); // logging is off by default, enable it
...
 
 const restClient = simpleRestClient('http://localhost:3000');
-const delayedRestClient = (type, resource, params) => new Promise((resolve, reject) => setTimeout(() => resolve(restClient(type, resource, params)), 1000));
+const delayedRestClient = (type, resource, params) => new Promise(resolve => setTimeout(() => resolve(restClient(type, resource, params)), 1000));
+
+const CustomLayout = withAppTitle('Example Admin')(Layout);
 
 render(
-    <Admin restClient={delayedRestClient}>
+    <Admin restClient={delayedRestClient} appLayout={CustomLayout}>
```
Comment (fzaninotto) [inline example/app.js]: Maybe we can add a `title` attribute to the `Admin` component to hide this complexity to the end user? Understanding HOCs just to change the title is a bit too much at that stage IMO.

File: src/mui/layout/Layout.js [hunk 4]
```diff
@@ -35,10 +35,6 @@ Layout.propTypes = {
     title: PropTypes.string.isRequired,
 };
 
-Layout.defaultProps = {
-    title: 'Admin on REST',
```
Comment (fzaninotto) [inline src/mui/layout/Layout.js]: Why did you remove that? It's still required as far as I understand.

Comment (djhi) [inline src/mui/layout/Layout.js]: Nope, specified as default on the title props of App component

**Comment by fzaninotto** [comment]
Great!
